### DISCUSSION TITLE
long backoff time when encountering a single 403

### DISCUSSION THREAD

**Discussion by bliu11** [PyGithub/PyGithub#2909] [General]
Can someone please help me out. 
I am pulling data on a list of users using get_user_by_id
However when a single user returns a 403, I'm getting a long backoff time.

Example
```
Request GET /user/****** failed with 403: Forbidden
Setting next backoff to 1784.060755s
```

**Comment by EnricoMi** [comment]
This is probably due to a secondary rate limit. There were too many requests and Github decided to forbid further access for a while. It provides a reset timestamp until which any request will be forbidden. In your example, that reset occurs in 1,784 seconds. Github waits defers the retry to eventually succeed the request.

https://docs.github.com/en/rest/using-the-rest-api/rate-limits-for-the-rest-api?apiVersion=2022-11-28#about-secondary-rate-limits
